src/server/pkg/hashtree/hashtree.go

```diff
@@ -525,22 +542,12 @@ func (h *hashtree) mergeNode(path string, srcs []HashTree) error {
 // - Code(e) is the error code of the first error encountered
 // - e.Error() contains the error messages of the first 10 errors encountered
 func (h *hashtree) Merge(trees []HashTree) error {
-	b, err := h.Serialize()
+	hcopy, err := h.clone()
 	if err != nil {
-		return errorf(Internal, "could not serialize hashtree before merge: %s", err)
+		return errorf(Internal, "could not copy hashtree before merge: %s", err)
 	}
 	if err = h.mergeNode("/", trees); err != nil {
-		htInterfaceTmp, deserializeErr := Deserialize(b)
-		if deserializeErr != nil {
-			return errorf(Internal, "could not deserialize hashtree (due to \"%s\") "+
-				"after merge error: %s", deserializeErr, err)
-		}
-		if htTmp, ok := htInterfaceTmp.(*hashtree); ok {
-			*h = *htTmp
-		} else {
-			return errorf(Internal, "could not convert deserialized hash tree after "+
-				"merge error: %s", err)
-		}
+		*h = *hcopy
```

Comment:
This also seems a little strange.  It seems to make more sense to perform the merge on `hcopy`, and only set `*h = *hcopy` if there's no error, as opposed to "resetting" `*h` if there's an error.